R/fill.R

```diff
@@ -108,5 +126,9 @@ fill.data.frame <- function(data, ..., .direction = c("down", "up", "downup", "u
     vec_fill_missing(col, direction = .direction)
   }
 
-  dplyr::mutate_at(data, .vars = dplyr::vars(any_of(vars)), .funs = fn)
+  dplyr::mutate(
+    data,
+    dplyr::across(any_of(vars), .fns = fn),
+    .by = {{ .by }}
```

Comment:
Passing straight through to dplyr means that if `.by` is wrong for some reason, then the error call shows `dplyr::mutate()` rather than `fill()`. I think that is worth it to not have to fiddle with `.by` in any way, because dplyr has some pretty unique handling for it. (See the tests for more info)